diff --git a/config.def.h b/config.def.h
index 9981ca3..2ecfe06 100644
--- a/config.def.h
+++ b/config.def.h
@@ -18,8 +18,13 @@ enum {
  * (first/last value is used as min/max zoom level)
  */
 static const float zoom_levels[] = {
-	 12.5,  25.0,  50.0,  75.0,
-	100.0, 150.0, 200.0, 400.0, 800.0
+       // set a low minimum zoom for shrink-to-fit
+         5.0,
+       // [round((0.25**(1/10))**x * 100, 1) for x in range(10, 0, -1)]
+        25.0,  28.7,  33.0,  37.9,  43.5,  50.0,  57.4,  66.0,  75.8,  87.1,
+       100.0,
+       // [round((0.25**(1/10))**x * 100, 1) for x in range(-1, -11, -1)]
+       114.9, 132.0, 151.6, 174.1, 200.0, 229.7, 263.9, 303.1, 348.2, 400.0
 };
 
 /* default slideshow delay (in sec, overwritten via -S option): */
@@ -32,7 +37,7 @@ static const double GAMMA_MAX   = 10.0;
 static const int    GAMMA_RANGE = 32;
 
 /* command i_scroll pans image 1/PAN_FRACTION of screen width/height */
-static const int PAN_FRACTION = 5;
+static const int PAN_FRACTION = 4;
 
 /* if false, pixelate images at zoom level != 100%,
  * toggled with 'a' key binding
@@ -48,10 +53,10 @@ static const bool ALPHA_LAYER = false;
 #ifdef _THUMBS_CONFIG
 
 /* thumbnail sizes in pixels (width == height): */
-static const int thumb_sizes[] = { 32, 64, 96, 128, 160 };
+static const int thumb_sizes[] = { 64, 128, 256 };
 
 /* thumbnail size at startup, index into thumb_sizes[]: */
-static const int THUMB_SIZE = 3;
+static const int THUMB_SIZE = 1;
 
 #endif
 #ifdef _MAPPINGS_CONFIG
@@ -63,7 +68,7 @@ static const keymap_t keys[] = {
 	{ 0,            XK_Return,        g_switch_mode,        None },
 	{ 0,            XK_f,             g_toggle_fullscreen,  None },
 	{ 0,            XK_b,             g_toggle_bar,         None },
-	{ ControlMask,  XK_x,             g_prefix_external,    None },
+	{ ControlMask,  XK_space,         g_prefix_external,    None },
 	{ 0,            XK_g,             g_first,              None },
 	{ 0,            XK_G,             g_n_or_last,          None },
 	{ 0,            XK_r,             g_reload_image,       None },
@@ -84,8 +89,6 @@ static const keymap_t keys[] = {
 	{ 0,            XK_M,             g_mark_range,         None },
 	{ ControlMask,  XK_m,             g_reverse_marks,      None },
 	{ ControlMask,  XK_u,             g_unmark_all,         None },
-	{ 0,            XK_N,             g_navigate_marked,    +1 },
-	{ 0,            XK_P,             g_navigate_marked,    -1 },
 	{ 0,            XK_braceleft,     g_change_gamma,       -1 },
 	{ 0,            XK_braceright,    g_change_gamma,       +1 },
 	{ ControlMask,  XK_g,             g_change_gamma,        0 },
@@ -100,12 +103,18 @@ static const keymap_t keys[] = {
 	{ 0,            XK_Right,         t_move_sel,           DIR_RIGHT },
 	{ 0,            XK_R,             t_reload_all,         None },
 
-	{ 0,            XK_n,             i_navigate,           +1 },
-	{ 0,            XK_n,             i_scroll_to_edge,     DIR_LEFT | DIR_UP },
 	{ 0,            XK_space,         i_navigate,           +1 },
-	{ 0,            XK_p,             i_navigate,           -1 },
-	{ 0,            XK_p,             i_scroll_to_edge,     DIR_LEFT | DIR_UP },
+	{ 0,            XK_period,        i_navigate,           +1 },
+	{ 0,            XK_period,        i_scroll_to_edge,     DIR_LEFT | DIR_UP },
 	{ 0,            XK_BackSpace,     i_navigate,           -1 },
+	{ 0,            XK_comma,         i_navigate,           -1 },
+	{ 0,            XK_comma,         i_scroll_to_edge,     DIR_LEFT | DIR_UP },
+	{ ControlMask,  XK_space,         g_navigate_marked,    +1 },
+	{ ControlMask,  XK_period,        g_navigate_marked,    +1 },
+	{ ControlMask,  XK_period,        i_scroll_to_edge,     DIR_LEFT | DIR_UP },
+	{ ControlMask,  XK_BackSpace,     g_navigate_marked,    -1 },
+	{ ControlMask,  XK_comma,         g_navigate_marked,    -1 },
+	{ ControlMask,  XK_comma,         i_scroll_to_edge,     DIR_LEFT | DIR_UP },
 	{ 0,            XK_bracketright,  i_navigate,           +10 },
 	{ 0,            XK_bracketleft,   i_navigate,           -10 },
 	{ ControlMask,  XK_6,             i_alternate,          None },
@@ -143,10 +152,11 @@ static const keymap_t keys[] = {
 static const button_t buttons[] = {
 	/* modifiers    button            function              argument */
 	{ 0,            1,                i_cursor_navigate,    None },
+	{ 0,            1,                i_scroll_to_edge,     DIR_LEFT | DIR_UP },
 	{ 0,            2,                i_drag,               DRAG_ABSOLUTE },
 	{ 0,            3,                g_switch_mode,        None },
-	{ 0,            4,                g_zoom,               +1 },
-	{ 0,            5,                g_zoom,               -1 },
+	{ 0,            4,                i_scroll,             DIR_UP },
+	{ 0,            5,                i_scroll,             DIR_DOWN },
 };
 
 #endif
